### ISSUE TITLE
Support for teledisk image format (TD0)

### ISSUE THREAD

**Issue by Pyrofer** [davidgiven/fluxengine#311]
The HP 200 series machines have a software archive that appears to be in teledisk format. This is old and only supported on vintage hardware itself. The TD0 files (example here http://www.hpmuseum.net/software/systest1.zip ) are rquired to get my HP 9826 working. 
I have only Greaseweazle hardware which natively supports only .scp (supercard Pro) file format.
Can Teledisk support be added for the HP 200 series machines please?

**Comment by ldkraemer** [comment]
The *.td0 images in that Teledisk .TD0 file are not good images.  I've had a look at both with the HxC Software
(Linux Version) and they are bad images.
![HP-td0](https://user-images.githubusercontent.com/36509186/126819310-d97c5af2-bf80-4aa1-8be5-f150e4286ec9.png)

There are easier ways to convert good *.TD0 Images to the format you want, versus having David jump through hoops to
add conversion code.  If you run Windows download and play with SAMDISK.  It is a good package and will convert *.TD0
files to several formats.
https://simonowen.com/samdisk/
https://simonowen.com/samdisk/formats/


Also the HxC Software will do the same conversions, plus give the image view as shown above.  Jeff's Software supports
Windows & Linux.  It works and can do more than just conversions.
https://hxc2001.com/download/floppy_drive_emulator/

Linux:  [url]https://sourceforge.net/p/hxcfloppyemu/code/HEAD/tree/[/url]

There are also the Imagedisk Utilities by David Dunfield.  His TD02IMD, IMDA, and IMDU utilities will create
*.IMD, *.RAW, and *.TXT files from *.TD0 Files.  You can get good information on the *.TD0's Image.
[url]http://dunfield.classiccmp.org/img/[/url]

If you are still not satisfied try out cpmtools built with libdsk.  John Elliott has DSKTRANS that will convert from about
any type Source Image to many different types Output images.  I know you don't care about the OLD CP/M Images
but they are supported.    [url]https://www.seasip.info/Unix/LibDsk/[/url]

$ man dsktrans

$ dsktrans -types
Disk image types supported:

   gotek      : Gotek 1440k disc image collection 
   gotek72    : Gotek 720k disc image collection 
   remote     : Remote LibDsk instance
   rcpmfs     : Reverse CP/MFS driver
   floppy     : Linux floppy driver
   dsk        : CPCEMU .DSK driver
   edsk       : Extended .DSK driver
   apridisk   : APRIDISK file driver
   copyqm     : CopyQM file driver
   tele       : TeleDisk file driver
   ldbs       : LibDsk block store
   ldbst      : LDBS (text form)
   sap        : SAP file driver
   qrst       : Quick Release Sector Transfer
   imd        : IMD file driver
   ydsk       : YAZE YDSK driver
   raw        : Raw file driver (alternate sides) 
   rawoo      : Raw file driver (out and out) 
   rawob      : Raw file driver (out and back) 
   myz80      : MYZ80 hard drive driver
   simh       : SIMH disc image driver
   nanowasp   : NanoWasp image file driver
   logical    : Raw file logical sector order
   jv3        : JV3 file driver
   dc42       : Disk Copy 4.2
   cfi        : CFI file driver


$ dsktrans -formats
Disk formats supported:

   pcw180     : PCW / IBM 180k
   cpcsys     : CPC System
   cpcdata    : CPC Data
   pcw720     : PCW / IBM 720k
   pcw1440    : PcW16 / IBM 1440k 
   ibm160     : IBM 160k (CP/M-86 / DOSPLUS)
   ibm320     : IBM 320k (CP/M-86 / DOSPLUS)
   ibm360     : IBM 360k (CP/M-86 / DOSPLUS)
   ibm720     : IBM 720k (144FEAT)
   ibm1200    : IBM 1.2M (144FEAT)
   ibm1440    : IBM 1.4M (144FEAT)
   acorn160   : Acorn 160k
   acorn320   : Acorn 320k
   acorn640   : Acorn 640k
   acorn800   : Acorn 800k
   acorn1600  : Acorn 1600k
   pcw800     : PCW 800k
   pcw200     : PCW 200k
   bbc100     : BBC 100k
   bbc200     : BBC 200k
   mbee400    : Microbee 400k
   mgt800     : MGT 800k
   trdos640   : TR-DOS 640k
   ampro200   : Ampro 40 track single-sided
   ampro400d  : Ampro 40 track double-sided
   ampro400s  : Ampro 80 track single-sided
   ampro800   : Ampro 80 track double-sided
   pcw1200    : PcW16 / IBM 1200k 
   mac400     : Macintosh GCR 400k
   mac800     : Macintosh GCR 800k
   myz80      : MYZ80 8Mb
   pcpm320    : IBM 320k (CP/M-86 / DOSPLUS)
   yaze4m     : yaze4m image
   yaze2m     : yaze2m image
   yaze1m     : yaze1m image
   yaze512    : yaze512 image
   yaze256    : yaze256 image
   jr01       : JR01  Junior 80 - DSDD 48 tpi 5.25" - 512 x 9
   hea1       : HEA1  Heath H89, Magnolia CP/M - SSDD 48 tpi 5.25" - 512 x 9
   lbb_40dsdd : LBB 5.25" DS DD 40T (5.25" DS DD 40x9 s/t)
   lbb_77dshd : LBB 5.25" DS HD 77T (5.25" DS HD 77T 17x512 s/t)
   act3       : ACT3  Action Computer Ent.  Alloy 66 - SSHD 8" - 512 x 16
   dav1       : DAV1  Davidge - DSDD 96 tpi 5.25" - 1024 x 5
   comc       : COMC  Compupro (Viasyn) - DSSD 8" - 128 x 26
   bee4       : BEE4  Microbee Systems - DSDD 3.5" - 512 x 10
   meg2       : MEG2  Megatel Quark 100 -	 DSDD 96 tpi 5.25" - 128 x 30
   portapak   : PortaPak 5.25" 80T (5.25" DS DD 80T 10x512 s/t)
   amsd       : AMSD  Amstrad PCW 9256, 9512 Disckit - DSDD 3.5" - 512 x 9
   mdsad350   : Northstar Micro-Disk System MDS-A-D 350 - SSDD 48 tpi 5.25" - 512 x 10
   mdsad175   : Northstar Micro-Disk System MDS-A-D 175 - SSDD 48 tpi 5.25" - 512 x 10
   mdsad87    : Northstar Micro-Disk System MDS-A1 - SSSD 48 tpi 5.25" - 256 x 10
   xer6       : XER6  Xerox 16/8 - DSDD 8" - 256 x 26
   xer5       : XER5  Xerox 16/8 - SSDD 8" - 256 x 26
   ibm2       : IBM2  IBM PC, CP/M-86 - DSDD 48 tpi 5.25" - 512 x 8
   sag1       : SAG1  Sage IV - DSDD 48 tpi 5.25" - 512 x 8
   xerf       : XERF  Xerox 16/8 Special - DSDD 48 tpi 5.25" - 512 x 8
   tek1       : TEK1  Tektronix 4170 - DSDD 48 tpi 5.25" - 512 x 8
   see2       : SEE2  Seequa Chameleon - DSDD 48 tpi 5.25" - 512 x 8
   oli1       : OLI1  Olivetti ETV300 - SSDD 48 tpi 5.25" - 256 x 18
   ibx2       : IBX2  IBEX 7301 - DSDD 8" - 256 x 26
   adv4       : ADV4  Advanced Digital Super 8 - DSDD 8" - 1024 x 8
   exi2       : EXI2  Exidy Sorcerer - SSDD 8" - 256 x 16
   exi1       : EXI1  Exidy Sorcerer - SSDD 48 tpi 5.25" - 256 x 16
   jad1       : JAD1  Jade - SSDD 8" - 128 x 48
   als2       : ALS2  Alspa - DSDD 8" - 1024 x 8
   als1       : ALS1  Alspa - SSDD 8" - 1024 x 8
   tos3       : TOS3  Toshiba T-250 - DSHD 8" - 256 x 26
   tos1       : 
   rc72       : PICCOLO RC702 -- DSDD 8" - 512 x 15
   rc03       : RC03 RC-700 Piccolo RCTELST - DSDD 48 tpi 5.25" - 512 x 9
   rc02       : RC-750 Piccolo - DSHD 96 TPI 5.25" - 1024 x 8
   rc01       : RC01 RC-702 Piccolo - DSDD 48 tpi 5.25" - 512 x 9
   sup5       : SUP5  Superbrain - SSDD 48 tpi 5.25" - 128 x 30
   sup4       : SUP4  Superbrain II - DSDD 96 tpi 5.25" - 512 x 10
   sup3       : SUP3  Superbrain QD - DSDD 48 tpi 5.25" - 512 x 10
   sup2       : SUP2  Superbrain 40 track - SSDD 48 tpi 5.25" - 512 x 10
   sup1       : SUP1  Superbrain JR - SSDD 48 tpi 5.25" - 512 x 10
   pie1       : PIE1  Pied Piper - DSDD 96 tpi 5.25" - 512 x 10
   bon3       : BON3  Bondwell 2 - SSDD 3.5" - 256 x 18
   bw14       : BON2  Bondwell 14 - DSDD 48 tpi 5.25" - 256 x 18
   bw12       : BON1  Bondwell 12 - SSDD 48 tpi 5.25" - 256 x 10
   xer2       : XER2  Xerox 820 - SSSD 48 tpi 5.25" - 128 x 18
   com9x      : COM9X  Compupro (Viasyn) 8/16 - DSDD 8" - 1024 x 8
   com8x      : COM8X  Compupro (Viasyn) 8/16 - SSDD 8" - 1024 x 8
   eag2       : EAG2  Eagle III, IV, V - DSDD 96 tpi 5.25" - 1024 x 5
   eag1       : EAG1  Eagle I, II - SSDD 96 tpi 5.25" -1024 x 5
   trs6       : TRS6  TRS-80, Pickles & Trout CP/M - SSDD 8" - 512 x 16
   trs5       : TRS5  TRS-80, Lifeboat CP/M (1024 bytes/sector) - SSDD 8" - 1024 x 8
   trs4       : TRS4  TRS-80 Lifeboat CP/M (256 bytes/sector) - SSDD 8" - 256 x 26
   necc       : NECC  NEC APC CP/M-86 - DSHD 8" - 256 x 26
   adv5       : ADV5  Advanced Digital TurboDOS 312K - DSDD 48 tpi 5.25" - 1024 x 4
   ims3       : IMS3  IMS 5000 TurboDOS - SSDD 8" - 1024 x 8
   zen8       : ZEN8  Zenith Z-100 - DSDD 48 tpi 5.25" - 512 x 8
   exo1       : EXO1  EXO - SSDD 8" - 512 x 16
   lob5       : LOB5  Lobo CP/M 2.2 - SSHD 8" - 256 x 30
   mai1       : MAI1  MAI Basic Four - DSDD 96 tpi 5.25" - 256 x 16
   san4       : SAN4  Sanyo MBC-3000 - DSDD 8" - 256 x 26
   san3       : SAN3  Sanyo MBC-2000 - SSDD 96 tpi 5.25" - 256 x 16
   san2       : SAN2  Sanyo  MBC-1250 - DSDD 96 tpi 5.25" - 256 x 16
   san1       : SAN1  Sanyo MBC-1000, MBC-1150 - DSDD 48 tpi 5.25" - 256 x 16
   mm2        : MM2  Micromint SB180 Native 2 - DSDD 96 tpi 5.25" - 1024 x 5
   mm1        : MM1  Micromint SB180 Native 1 - DSDD 48 tpi 5.25" - 512 x 10
   mor8       : MOR8  Morrow 5/11/34 - SSDD 48 tpi 5.25" - 1024 x 5
   mor7       : MOR7  Morrow MD3...CP/M Plus - DSDD 48 tpi 5.25" - 1024 x 5
   mor6       : MOR6  Morrow TurboDOS - DSDD 48 tpi 5.25" - 1024 x 4
   mor5       : MOR5  Morrow 8 (1024 bytes/sector) - DSDD 8" - 1024 x 8
   mor4       : MOR4  Morrow 8 (1024 bytes/sector) - SSDD 8" - 1024 x 8
   mor3       : MOR3  Morrow 8 (512 bytes/sector) - SSDD 8" - 512 x 15
   mor2       : MOR2  Morrow MD3, 5, 11, 16, 34 - DSDD 48 tpi 5.25" - 1024 x 5
   mor1       : MOR1  Morrow MD2 - SSDD 48 tpi 5.25" - 1024 x 5
   meg1       : MEG1  Megatel Quark 100 -	 SSDD 48 tpi 5.25" - 512 x 10
   bon3       : BON3  Bondwell 2 - SSDD 3.5" - 256 x 18
   bon2       : BON2  Bondwell 14 - DSDD 48 tpi 5.25" - 256 x 18
   osb8       : OSB8  Osborne Vixen - DSDD 48 tpi 5.25" - 1024 x 5
   osb3       : OSB3  Osborne Executive - SSDD 48 tpi 5.25" - 1024 x 5
   osb2       : OSB2  Osborne 1 - SSDD 48 tpi 5.25" - 1024 x 5
   osb1       : OSB1  Osborne 1 - SSSD 48 tpi 5.25" - 256 x 10
   phi2       : PHI2  Philips PC-2012 - DSDD 96 tpi 5.25" - 256 x 16
   phi1       : PHI1  Philips PC-2010 - SSDD 48 tpi 5.25" - 256 x 16
   pmc2       : PMC2  PMC Micromate - DSQD 96 tpi 5.25" - 1024 x 5
   pmc1       : PMC1  PMC Micromate - DSDD 48 tpi 5.25" - 1024 x 5
   eps2       : EPS2  Epson QX-10 - DSDD 48 tpi 5.25" - 256 x 16
   eps1       : EPS1  Epson QX-10 - DSDD 48 tpi 5.25" - 512 x 10
   spe2       : SPE2  Spectravideo SV-328 - DSDD 48 tpi 5.25" - 256 x 17
   spe1       : SPE1  Spectravideo 318/328 - SSDD 48 tpi 5.25" - 256 x 17
   son1       : SON1  Sony SMC-70 - SSDD 3.5" - 256 x 16
   sag2       : SAG2  Sage IV - DSDD 96 tpi 5.25" - 512 x 8
   sag1       : SAG1  Sage IV - DSDD 48 tpi 5.25" - 512 x 8
   tel4       : TEL4  TeleVideo 8nn/TPC-1 CP/M - DSDD 48 tpi 5.25" - 256 x 18
   vis1       : VIS1  Visual 1050 - SSDD 96 tpi 5.25" - 512 x 10
   alt5       : ALT5  Altos Series 5 - DSDD 96 tpi 5.25" - 512 x 9
   col5       : COL5  Coleco ADAM, 360K - DSDD 48 tpi 5.25" - 512 x 9
   col1       : COL1  Coleco ADAM, 40 track - SSDD 48 tpi 5.25" - 512 x 8
   dec3       : DEC3  DEC VT-180 - SSDD 48 tpi 5.25" - 512 x 9
   alt3       : ALT3  Altos - SSDD 8" - 512 x 15
   alt1       : ALT1  Altos - SSSD 8" - 128 x 26
   big4       : BIG4  Big Board SWP 1024 bytes/sector - SSDD 8" - 1024 x 9
   big3       : BIG3  Big Board 1024 bytes/sector - SSDD 8" - 1024 x 9
   big2       : BIG2  Big Board (512 bytes/sector) - DSDD 8" - 512 x 15
   big1       : BIG1  Big Board (512 bytes/sector) - SSDD 8" - 512 x 15
   big0       : BIG0  Big Board (128 bytes/sector) - SSSD 8" - 128 x 26
   acn1       : ACN1  Acorn CP/M on Acorn Z80 second processor. - 256 x 10
   sie2       : Siemens PG-675 DSDD 48 tpi 5.25" - 512 x 9
   sin4       : Sinclair Spectrum ZX Lexor CP/M - DSDD 48 tpi 5.25"
   sin3       : Sinclair Spectrum ZX - SSDD 48 tpi 5.25"
   sin2       : Sinclair ZX Spectrum+3 CP/M 3.0 - SSDD 3.5"
   elwro      : Elwro Computer CP/M 3.0 - DSDD 3.5"
   sin1       : Sinclair ZX Spectrum+3 DOS - DSDD 3.5"
   zqq        : Zorba qq - DSDD 48 tpi 5.25"
   zor2       : ZOR2  Zorba - DSDD 96 tpi 5.25" - 512 x 10
   zor1       : ZOR1  Zorba - DSDD 48 tpi 5.25" - 512 x 10
   trsf       : TRSF  TRS-80 II/12/16 Aton CP/M  - DSHD 8" - 1024 x 8
   trse       : TRSE  TRS-80 II/12/16 Aton CP/M - SSHD 8" - 1024 x 8
   a1         : A1  Generic CP/M - SSSD 8" - 128 x 26
   trsw       : TRS-80 Model 4,4P Montezuma System 400K - SSDD 96 tpi 3.5"
   trsv       : TRS-80 Model 4,4P Montezuma Super Data 880K - DSDD 96 tpi 5.25"
   trsu       : TRS-80 Model 4,4P Montezuma Super Data 440K - SSDD 96 tpi 5.25"
   trst       : TRS-80 Model 4,4P Montezuma Super Data 440K - DSDD 48 tpi 5.25"
   trss       : TRS-80 Model 4,4P Montezuma Super Data 220K - SSDD 48 tpi 5.25"
   trsr       : TRS-80 Model 4,4P Montezuma Extend System 790K - DSDD 96 tpi 5.25"
   trsq       : TRS-80 Model 4,4P Montezuma Extend System 390K - SSDD 96 tpi 5.25"
   trsp       : TRS-80 Model 4,4P Montezuma Extend System 390K - SSDD 48 tpi 5.25"
   trso       : TRS-80 Model 4,4P Montezuma Extend System 190K - SSDD 48 tpi 5.25"
   trsn       : TRS-80 Model 4,4P Montezuma Data 800K - DSDD 96 tpi 5.25"
   trsm       : TRS-80 Model 4,4P Montezuma System 710K - DSDD 96 tpi 5.25"
   trsl       : TRS-80 Model 4,4P Montezuma Data 400K - SSDD 96 tpi 5.25"
   trsk       : TRS-80 Model 4,4P Montezuma System 350K - SSDD 96 tpi 5.25"
   trsj       : TRS-80 Model 4,4P Montezuma Data 400K - DSDD 48 tpi 5.25"
   trsi       : TRS-80 Model 4,4P Montezuma System 350K - DSDD 48 tpi 5.25"
   trsh       : TRS-80 Model 4,4P Montezuma Data 200K - SSDD 48 tpi 5.25"
   trsg       : TRS-80 Model 4,4P Montezuma System 170K - SSDD 48 tpi 5.25"
   ampro800   : Ampro800 - DSDD 96 tpi 3.5"
   amp6       : Ampro - DSDD 96 tpi 3.5"
   amp5       : Ampro - SSDD 96 tpi 3.5"
   amp4       : Ampro - DSDD 96 tpi 5.25"
   amp3       : Ampro - SSDD 96 tpi 5.25"
   amp2       : Ampro - DSDD 48 tpi 5.25"
   amp1       : Ampro - SSDD 48 tpi 5.25"
   kay3       : KAY3  Kaypro 2X/4/10 (Alternate) - DSDD 48 tpi 5.25" - 512 x 10
   kay2       : KAY2  Kaypro 2X/4/10 - DSDD 48 tpi 5.25" - 512 x 10
   kpiv       : Kaypro 2X/4/10 - DSDD 48 tpi 5.25"
   kpii       : KAY1  Kaypro II/2 - SSDD 48 tpi 5.25" - 512 x 10
   kay1       : KAY1  Kaypro II/2 - SSDD 48 tpi 5.25" - 512 x 10
   ibm3740    : IBM3740 SS SD 77T 8" 26x128 b/s

There isn't much sense for Mr. Given to duplicate the work of previous Software Authors that are providing free software.

If you want to use the Imagedisk Utilities, you can run DOSBox and execute a batch file to get the conversions automated.

A.BAT contains
TD02IMD systest1.td0 > SYSTEST1.TXT
IMDA SYSTEST1.IMD >> SYSTEST1.TXT
IMDU SYSTEST1.IMD SYSTEST1.RAW /B /E /D >> SYSTEST1.TXT

Will get you a SYSTEST.IMD file along with a STSTEST1.RAW (Sector Dump) and information on the *.TD0 file.

Note:  A *.TD0 file is a container just like *.IMD is a slightly different container for the data.

Glad to offer any assistance I can.....

Larry

**Comment by Pyrofer** [comment]
Thank you for the long and detailed reply. I was actually asked to log this issue here after a conversation on reddit, so that's why it's here.
As for your reply. Lots to take in (which is good?). Firstly and most importantly you say
"The *.td0 images in that Teledisk .TD0 file are not good images. I've had a look at both with the HxC Software
(Linux Version) and they are bad images."

Is that suggesting that they are corrupt or damaged in some way? ie, I can not get a good disk written from them?

My goal here is simply to make some boot disks for an HP machine. I already tried HxC which allowed me to load in the TD0 and export an SCP (Supercar Pro, the format my Greaseweazle supports). I made the disk from that scp and it does not work. I was suspecting the floppy drive but had no assurance the conversion and writing had gone well.
From what you wrote I am thinking that the faulty .TD0 was the problem and converting it to SCP didn't help, making a faulty disk?

I've already got HxC and will gram samdisk to play with, but if there is a suspicion the images are broken anyway I might be wasting my time?

**Comment by ldkraemer** [comment]
If you have a look at the systest2.td0 file it is only 60KB in size, and systest1.td0 is 225.9KB.  The second one
could be good, but dsktrans would not extract it properly and neither would IMD (Imagedisk).  I tried both
then tried HxC to view the image.  HxC should show good pie shaped slices for the sectors and not
variable as this file is.  I wouldn't expect it to boot.  If you can download some better TD0 or IMD files you will
get it booting.  

I forgot to mention that dsktrans and SAMDisk will write direct to the floppy if you have the hardware attached.
Does not work with USB floppy drives.

Do you know the layout of the floppy 80 Tracks of xxx bytes per sector, and xx sectors per track single or double sided?
You should be able to make a close guess of the size by multiplying those parameters.

SAMDisk should convert from TD0 — Sydex TeleDisk (various) to raw — Raw sector dump in suitable regular format.

I've used it by running wine in Linux to extract *.RAW files.  This works in Linux.
[code]
$ wine SAMdisk mykpiiboot1.scp mykpiiboot1.raw
[/code]

Try to locate a couple good *.TD0 or *.IMD files.

Thanks.

Larry

**Comment by Pyrofer** [comment]
Well, there is a CP/M disk image here http://www.hpmuseum.net/software/CPM68K.TD0
It would also be fun to try the games disk here http://www.hpmuseum.net/software/200GAMES.zip
They have quite a few, I was mainly interested in a boot disk / language such as basic.
This is the museums boot+basic disk and is a more recent image so should be a good one? http://www.hpmuseum.net/software/BAS5BOOT.TD0

Thanks again for the help.

**Comment by ldkraemer** [comment]
I've downloaded those 4 files and extracted the *.TD0 files.  Not sure what version of Teledisk should work to
write these files, if it's possible to write them.  Probably will take several attempts to reproduce them.  I do see
lots of problems.  Here is just one example of what I am finding

IMageDisk Utility 1.18 / Mar 07 2012
IMD TD 1.5 3.5 LD MFM S-step, 2 sides 1/01/2004 11:54:44

HP Series 200 games

Assuming 1:1 for Binary output
 0/0 250 kbps DD  16x256            <<------------------256 byte sectors and 16 of them in the following order.
      1   9   2   10  3   11  4   12  5   13  6   14  7   15  8   16    <<----------------------------|
      D   D00 D00 D00 D   D00 D   D00 D   D00 D00 D00 D00 D00 D00 D00
 0/1  4   12  5   13  6   14  7   15  8   16  1   9   2   10  3   11 
      D20 D   D   D20 D   D   D20 D   D   D20 D   D   D   D20 D   D  
 1/0  16  1   9   2   10  3   11  4   12  5   13  6   14  7   15  8      <<------------But this track is different
      D   D   D20 D   D   D20 D   D   D20 D   D   D20 D   D   D20 D  
 1/1  2   10  3   11  4   12  5   13  6   14  7   15  8   16  1   9      <<------------And so is this one
      D20 D   D   D20 D   D   D20 D   D   D20 D   D   D20 D   D   D  
 2/0  8   16  1   9   2   10  3   11  4   12  5   13  6   14  7   15      <<-----------And on and on and on
      D   D20 D20 D   D   D20 D   D   D20 D   D   D20 D   D   D20 D  
 2/1  3   11  4   12  5   13  6   14  7   15  8   16  1   9   2   10 
      D20 D   D   D20 D   D   D20 D   D   D20 D   D   D   D20 D   D  
 3/0  D   D20 D20 D   D20 D20 D   D20 D20 D   D20 D20 D   D   D20 D20
 3/1  1   9   2   10  3   11  4   12  5   13  6   14  7   15  8   16 
      D20 D20 D   D20 D20 D   D20 D20 D   D20 D20 D   D20 D20 D   D20
 4/0  D   D20 D20 D   D20 D20 D   D20 D20 D   D20 D20 D   D20 D20 D  
 4/1  D20 D   D20 D20 D   D20 D20 D   D20 D20 D   D20 D20 D   D20 D20
 5/0  2   10  3   11  4   12  5   13  6   14  7   15  8   16  1   9  
      D   D20 D20 D   D20 D20 D   D20 D20 D   D20 D20 D   D20 D20 D20
 5/1  D20 D   D20 D20 D   D20 D20 D   D20 D20 D   D20 D20 D   D   D20
...
...
...
76/0  D00 D00 D00 D00 D00 D00 D00 D00 D00 D00 D00 D00 D00 D00 D00 D00
76/1  D00 D00 D00 D00 D00 D00 D00 D00 D00 D00 D00 D00 D00 D00 D00 D00
77/0 250 kbps DD  9x512                      <<-----------------Now changed to 512 Byte Sectors and 9 of them
      1   2   3   4   5   6   7   8   9  
      DF6 DF6 DF6 DF6 DF6 DF6 DF6 DF6 DF6
77/1  DF6 DF6 DF6 DF6 DF6 DF6 DF6 DF6 DF6
78/0  DF6 DF6 DF6 DF6 DF6 DF6 DF6 DF6 DF6
78/1  DF6 DF6 DF6 DF6 DF6 DF6 DF6 DF6 DF6
79/0 250 kbps DD  16x256                     <<-----------------Then back to 256 Byte Sectors and 16 of these 
      99  100 101 102 103 104 105 106 107 108 109 110 111 112 97  98 
      DFF DFF DFF DFF DFF DFF DFF DFF DFF DFF DFF DFF DFF DFF D   D  
79/1  DFF DFF DFF DFF DFF DFF DFF DFF DFF DFF DFF DFF DFF DFF D   D  
160 tracks(80/80), 2532 sectors (1175 Compressed)


The CP/M Image tells me there are mixed sector sizes within one track.

TD 1.5 3.5 LD MFM S-step, 1 sides 6/05/2012 17:23:16
HP 200 SERIES
CP/M FOR 68K
MUST BOOT FROM 9121
Cannot do mixed sector size within track


Just glad I'm not the Administrator of this mess.  It's the same for all four Images.
But, perhaps the correct version of Teledisk can/will write them correctly.  

Note, the fifth byte of the Teledisk Image, when viewed as a HEX file should be the version it was created with,
but this doesn't hold true for later versions as they all show the same 5.

00000000   54 44 00 4F  15 00 04 80  00 02 D8 77  DD 49 1B 00  TD.O.......w.I..

I just don't see anyway possible they can be reconstructed as a Bootable image and used.


Larry

**Comment by Pyrofer** [comment]
See, this is why I am having issues. If I can't even be sure the disk images are good how can I test the hardware!
I tried with another floppy drive and got similar results. Examining the original floppy drive reveals a standard TM100-2A IBM floppy, that appears to work. Belt is ok, motors spin and the head jiggles when you power up. Disk spins and it finds "something" and starts to load. Then it sits spinning for ages without the head advancing until it fails with random disk related (cant find sector, crc etc) errors.

The drive seems to move right and changing the floppy didn't get better results so I am going to re-assemble the machine and say the floppy drive is ok and it's the images that suck. Now I need somebody with an original 9826 boot disk to make a working image for me.

**Comment by davidgiven** [comment]
It's actually done!

Well, sort of. The images do look a bit weird to me; they're 70-track 18-sector 256-bytes-per-sector images, but each one is missing sector 16 (they're numbered 0-17). This might be intentional, as I've found an English text string split between sectors 15 and 17. And there's only just enough space on each track for the 17 sectors present. However, the images _do_ seem to be syntactically valid. (I've only looked at the two systest images for the 9826.)

I've added an HP 9121 profile, but I've had to guess at pretty much all the numbers, so it may or may not work.

Also, I've discovered and fixed a massive memory corruption bug in the FluxEngine client core code, so that's good!

It's in the `teledisk` branch. PTAL? https://github.com/davidgiven/fluxengine/pull/312

```
fluxengine write hp9121 -i systest1.td0 -d drive:0
```
